### DISCUSSION TITLE
Unable to load second to last frame of entire current video

### DISCUSSION THREAD

**Discussion by jannhan** [talmolab/sleap#1895] [Help!] [answered]
I'm trying to train a single animal model and am able to generate predictions on random frames or a single frame, but an error occurs when I try to generate predictions for the entire video that says it is unable to load the second to last frame of the video. This happens for multiple different videos I have tried, and it is always the second to last frame.

**Comment by roomrys** [comment, marked as answer]
Hi @jannhan,

Are you getting a message similar to:
```
KeyError: "Unable to load frame X from MediaVideo"
```
?

If so, we have seen a few problems like this occur when the video is not [reliably seekable](http://blog.loopbio.com/video-io-1-introduction.html). Try [reencoding your video](https://sleap.ai/help.html#does-my-data-need-to-be-in-a-particular-format) and let us know how that works.

---
#### Update: If reencoding doesn't work, and results in `moov atom not found`
As I was helping a user, we found that they had just copied the contents of a zipped file over without unzipping which resulted in first the `KeyError`, and then, when attempting to reencode, a `moov atom not found` error. After retrying but with **unzipping first**, we were able to get everything running smoothly.

---

Thanks,
Liezl

P.S. Also linking similar problems here for reference (namely KeyError "Unable to load frame" causing SLEAP to crash when predicting
#767, Unable to load frame #366, KeyError when trying to run inference #630, Encoding Multiple Videos using Batch File #719, Prediction all frames in a video and processed to 99% and error #742).

**Comment by jannhan** [comment]
Reencoding the video worked, thank you so much for your quick reply!
